cranelift/codegen/src/isa/x64/lower.isle

```diff
@@ -1749,6 +1749,10 @@
             (gpr_c Gpr (put_in_gpr c)))
            (with_flags (x64_test size gpr_c gpr_c) (cmove_from_values ty (CC.NZ) x y))))
 
+(rule -2 (lower (has_type ty (select c @ (value_type $I128) x y)))
+      (let ((cond_result IcmpCondResult (cmp_zero_i128 (CC.Z) c)))
+        (select_icmp cond_result x y)))
```

Comment:
If I'm not mistaken, I think you're fixing #5199 as well here.

Reply:
Updated description, thanks!